src/index.js

```diff
@@ -70,8 +78,85 @@ class HMIApp extends React.Component {
     handleDDToggle(){
         store.dispatch(setDDState(!this.props.dd));
     }
+    pickResolution(event) {
+        var match = event.target.value.match(/(\d+)x(\d+) Scale (\d+.?\d*)/);
+        var capability = {
+            systemCapabilityType: 'VIDEO_STREAMING',
+            videoStreamingCapability: {
+                scale: parseFloat(match[3]),
+                preferredResolution: {
+                    resolutionWidth: parseInt(match[1]),
+                    resolutionHeight: parseInt(match[2])
+                }
+            }
+        }
+        this.setState({ resolution: event.target.value, scale: match[3] });
+
+        bcController.onSystemCapabilityUpdated(capability, this.props.activeAppId);
+    }
+
+    onTouchEvent(type, event) {
+        if (!this.videoRect) {
+            var video = document.getElementById('navi_stream');
+            this.videoRect = video.getBoundingClientRect();
+        }
+
+        if (event.pageX < this.videoRect.left
+            || event.pageX > this.videoRect.right
+            || event.pageY < this.videoRect.top
+            || event.pageY > this.videoRect.bottom) {
+            return;
+        }
+
+        uiController.onTouchEvent(type, [{
+            id: 0,
+            ts: [ parseInt(event.timeStamp) ],
+            c: [{
+                x: event.pageX - this.videoRect.x,
+                y: event.pageY - this.videoRect.y
+            }]
+        }]);
+    }
+
+    onTouchBegin(event) {
+        this.touchInProgress = true;
+        this.onTouchEvent('BEGIN', event);
+    }
+
+    onTouchMove(event) {
+        if (!this.touchInProgress) {
+            return;
+        }
+
+        this.onTouchEvent('MOVE', event);
+    }
+
+    onTouchEnd(event) {
+        if (!this.touchInProgress) {
+            return;
+        }
+
+        this.touchInProgress = false;
+        this.onTouchEvent('END', event);
+    }
+
     render() {
         const themeClass = this.state.dark ? 'dark-theme' : 'light-theme';
+        var videoStyle = { position: 'absolute', width: 960, height: 600, top: 75, left: 0, backgroundColor: 'transparent', objectFit: 'fill' };
```

Comment:
Streaming in any template is interesting. I dont mind it, but I think we should add a basic Full Screen Nav Map as a supported template. It can just be an empty page with the header at the top. If we have time we can add the nav buttons + text.

Reply:
Good point, we definitely need a NAV template, but the actual video DOM element needs to be in this component so that it won't be removed when the route changes and the template component is unmounted.

I have added this template in commit 8dbd4bb